Readability.js

```diff
@@ -155,6 +156,21 @@ Readability.prototype = {
   // These are the classes that readability sets itself.
   CLASSES_TO_PRESERVE: [ "page" ],
 
+  // These are the list of HTML entities that need to be escaped.
```

Comment:
This seems like it belongs in `JSDOMParser.js` directly, if that's where the issue is. Perhaps it's just a question of expanding `https://github.com/mozilla/readability/blob/master/JSDOMParser.js#L32` ?

Reply:
No, I believe the code in `JSDOMParser` already corrects (in fact I think I should use it instead of mine). However, I think we still need to decode HTML entities in `getArticleMetadata` because :

1. **Some web pages encode their metadata content twice.**

   For example, in `telegraph` their description content in source looks like this :

   ```
   ... arrest at Mugabe's &amp;quot;Blue House&amp;quot; ...
   ```

   which when decoded will become like this :

   ```
   ... arrest at Mugabe's &quot;Blue House&quot; ...
   ```

2. **There is different behavior between `jsdom` and `JSDOMParser`.**

   In `lazy-image-2` the description is like this :

   ```
   ... That&amp;#039;s what you&amp;#039;re about to see here ...
   ```

   which when decoded will become like this :
   
   ```
   ... That&#039;s what you&#039;re about to see here ...
   ```

   However, in `JSDOMParser` the [decoding process](https://github.com/mozilla/readability/blob/668a3a1010baab7c717ef5f04dea682fd3c78843/JSDOMParser.js#L60) has two steps: 1) first replace entities with common names; then 2) replace entities with numeric code. Thanks to this, `JSDOMParser` decode above description into this :

   ```
   ... That's what you're about to see here ...
   ```

   When generating test we are using `JSDOMParser`, while in testing we use both `jsdom` and `JSDOMParser`. However, thanks to this different behavior the test with `jsdom` will fail.

---

With that said, what do you think should I do ? I don't know which behavior is correct between `jsdom` and `JSDOMParser`, however I believe it's still worth it to decode HTML entities to prevent case like `telegraph`.